### PR TITLE
feat: return instantiated variable mapping in `Scheme.instantiate`

### PR DESCRIPTION
**Pull request by dghosef**
This PR updates `Scheme.instantiate` to return a type variable mapping from the original generalized variable to the instantiated variable. It also updates all current uses of `Scheme.instantiate` to ignore it.

This is useful for structs.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Question: Should we return the map or a Substitution?

**Comment by dghosef** [comment]
> Question: Should we return the map or a Substitution?

The map because
1) It is easy to construct the substitution when we need it
2) Having access to the raw map is useful in instantiate because sometimes I want to look up individual type variables. e.g. to find what the instantiated region variable is. `Substitution.m` gives a `kindedTypeVarSym -> Type` map but it is usefule to lookup a `kindedTypeVarSym -> Type.Var` map

**Comment by magnus-madsen** [comment]
@mlutze If you agree, you can merge.

File: main/src/ca/uwaterloo/flix/language/ast/Scheme.scala [hunk 1]
```diff
@@ -45,18 +45,19 @@ object Scheme {
...
     // Compute the fresh variables taking the instantiation mode into account.
     //
-    val freshVars = sc.quantifiers.foldLeft(Map.empty[Int, Type.Var]) {
+    val substMap = sc.quantifiers.foldLeft(Map.empty[Symbol.KindedTypeVarSym, Type.Var]) {
       case (macc, tvar) =>
         // Determine the rigidity of the fresh type variable.
-        macc + (tvar.id -> Type.freshVar(tvar.kind, loc, tvar.isRegion, Ast.VarText.Absent))
+        macc + (tvar -> Type.freshVar(tvar.kind, loc, tvar.isRegion, Ast.VarText.Absent))
     }
+    val freshVars = substMap.map(s => s._1.id -> s._2)
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/ast/Scheme.scala:60]: The `_1` stuff always makes me sad.

Can we do `substMap.map { (k, v) => k.id -> v }` instead?

**Review by mlutze — APPROVED** [review]

**Pull request merged**